gpflow/likelihoods/base.py

```diff
@@ -340,10 +397,10 @@ def _variational_expectations(self, Fmu, Fvar, Y):
         :param Y: observation Tensor, with shape [..., latent_dim]:
         :returns: variational expectations, with shape [...]
         """
-        return tf.reduce_sum(
-            ndiagquad(self._scalar_log_prob, self.num_gauss_hermite_points, Fmu, Fvar, Y=Y),
-            axis=-1,
-        )
+        if self.latent_dim is None:
+            self.latent_dim = int(Fmu.shape[-1])
```

Comment:
As you know, this will only work in eager mode. Would the `NDiagGHQuadrature(latent_dim,...)` also work when `latent_dim` is obtained using `tf.shape(Fmu)[-1]`?

Reply:
Not sure. Need to test it.